packages/qwik-city/buildtime/markdown/rehype.ts

```diff
@@ -66,14 +66,16 @@ function exportContentHeadings(mdast: Root) {
 
   visit(mdast, 'element', (node: any) => {
     const level = headingRank(node);
-    if (level && node.properties && !hasProperty(node, 'id')) {
+    if (level && node.properties) {
       const text = toString(node);
-      const id = slugs.slug(text);
-      node.properties.id = id;
+
+      if (!hasProperty(node, 'id')) {
+        node.properties.id = slugs.slug(text);
+      }
 
       headings.push({
         text,
-        id,
+        id: node.properties.id,
```

Comment:
This is needed because if the user passes a plugin like `rehypeAutolinkHeadings`, the ID is already added to the headings, so this is going to be false and headings would not be exported at all. With this change, the ID is added if it's missing, so it's guaranteed that headings is always exported.